src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1923,11 +1923,11 @@ private void GetSlotsToMarkAsNotNullable(BoundExpression operand, ArrayBuilder<i
 
                 return;
             }
-
-            bool shouldUpdateType(TypeSymbol operandType)
-                => !(operandType is null) && (!operandType.IsValueType || operandType.IsNullableType());
         }
 
+        private static bool IsTrackableType(TypeSymbol operandType)
```

Comment:
>IsTrackableType [](start = 28, length = 15)

Consider adding a doc comment explaining what does it mean for a type to be trackable. Or consider using different, more descriptive name. "PossiblyNullableType"?